### ISSUE TITLE
Output/classes/products directories do not handle parentheses

### ISSUE THREAD

**Issue by alexargo** [appsquickly/frankencover.it#10]
Hmmm... i'm noticing something else weird... it's not actually putting output where it says it will when I use on my project. Your build script for Typhoon is outputting the report however.

     Alexs-Mac-mini:Common Alex$ groovy http://frankencover.it/with --source-dir "${PWD}/Common"
 
     ▸ Code Coverage /Users/Alex/Code/jenkins/workspace/LibCommon(iOS)/Libraries/ObjectiveC/Common/Common/
     ✗ lines........: 6.4% (376 of 5845 lines)
     :bar_chart: full report..: build/reports/coverage/index.html

     Alexs-Mac-mini:Common Alex$ ls build/reports/
     junit.xml

I tried to specify directory that already exists with other files in it: 
     Alexs-Mac-mini:Common Alex$ groovy http://frankencover.it/with --source-dir "${PWD}/Common" --output-dir "${PWD}/build"

     ▸ Code Coverage /Users/Alex/Code/jenkins/workspace/LibCommon(iOS)/Libraries/ObjectiveC/Common/Common/
     Caught: java.lang.RuntimeException:

     ✗ Coverage data not found
     ☐ To see how to set build flags, visit http://frankencover.it
     ☐ Exercise your code with tests or otherwise.
     ☐ Run this script again.

     java.lang.RuntimeException:

     ✗ Coverage data not found
     ☐ To see how to set build flags, visit http://frankencover.it
     ☐ Exercise your code with tests or otherwise.
     ☐ Run this script again.

     at ReportParser.<init>(with:203)
     at ReportGenerator.printSummary(with:166)
     at ReportGenerator.generate(with:153)
     at ReportGenerator$generate.call(Unknown Source)
     at FrankenCliReader.execute(with:62)
     at FrankenCliReader$execute.call(Unknown Source)
     at with.run(with:13)

After that I tried a directory that doesnt' exist and one that does exists but is empty all with the same output.

**Comment by jasperblues** [comment]
Can you try running with the `--debug` option?

**Comment by alexargo** [comment]
My debug output is here for case where things seem to run successfully but there is no report generated: https://gist.github.com/alexargo/aa8291dad87f3687484e

I'm thinking the most likely source of the error is the space near the end of line 12 where the source directory is getting appended.

Also, one more point of confusion: It looks like it's attempting to output to the specified source directory (minus that random space) but the command just reports a relative "build/reports/coverage/index.html" with the default input. I was looking in the wrong place. It's not in either though (again, I think due to that space). It may be useful to report the full path to the directory of the console output.

**Comment by jasperblues** [comment]
If you want to try a fix you can download the script and run it locally. Then just edit with an ordinary text editor like Atom. (Or IntelliJ has code-completion, syntax highlighting, error checking, etc).

**Comment by alexargo** [comment]
I'll take a look later. It's getting pretty late here. Thanks for looking into it. Saw your title change: I don't think it's that it doesn't handle output directories with spaces... either there shouldn't be one there, or I need look into what the genhtml command line parameters ... now I'm guessing the space is just you separating two command line arguments.

**Comment by jasperblues** [comment]
@alexargo OK, thanks for your help. If you'd like to try to debug later, great. Otherwise is it possible to send a sample project that produces the problem, and we'll fix it for you?

**Comment by siddharthan64** [comment]
@alexargo  How did you handle the white space character issue?

**Comment by jasperblues** [comment]
@siddharthan64,

@alexargo Fixed up an issue with the scripts. You could compare commits to see.

**Comment by alexargo** [comment]
@siddharthan64 My issue ended up being with parenthesis in my paths. I believe i tested with spaces and it doesn't work. Just curious but what's the full command you are trying to run with?

My commit is here if you'd like to see what I did: https://github.com/alexargo/frankencover.it/commit/eed9f09c0f08b594baf196efb70e74c39d2f3235

**Comment by siddharthan64** [comment]
The issue I had was cmd

${genHtmlCmd} -o ${escapeSpecialCharacters(reportLocation.absolutePath)} --prefix ${escapeSpecialCharacters(reportConfig.prefix())} ${escapeSpecialCharacters(coverageInfoFile)}

Since I was running it on my Jenkins & with this script I had to use the absolute path of the project location and coverage file. The absolute path had a folder with space character. I decided to update the script by running the genhtml cmd on the shell instead of the script. 

the new cmd looked like genhtml -o ./ProjectFolder/myProject.xcodeproj --prefix ....
